### PR TITLE
Mjhsieh

### PR DESCRIPTION
**Pull request by mhsieh**

### CODE REVIEW

File: ports/pacbio/blasr/Makefile [hunk 1]
```diff
@@ -46,8 +46,8 @@ $(_WRKSRC)/utils/bax2bam/build/Makefile:
               -DHDF5_RootDir=$(HDF5_ROOT) \
               -DPacBioBAM_INCLUDE_DIRS=$(PREFIX)/include \
               -DPacBioBAM_LIBRARIES=$(PREFIX)/lib/libpbbam.$(DYLIB) \
-              -DHTSLIB_INCLUDE_DIRS=$(PREFIX)/include \
-              -DHTSLIB_LIBRARIES=$(PREFIX)/lib/libhts.$(DYLIB) \
+              -DHTSLIB_INCLUDE_DIRS=$(PREFIX)/src/htslib/include \
+              -DHTSLIB_LIBRARIES=$(PREFIX)/src/htslib/lib/libhts.a \
```
Comment (pb-cdunn) [inline ports/pacbio/blasr/Makefile]: This is a bit dangerous.

First, you can end up with an archive built into 2 different shared libraries, so the same library end up in a program twice. Duplication of the symbols would be prevented by the dynamic loader, but static initializers would be called twice. This happened at Amazon.

Second, OSX does not (practically) support static linking.

**Comment by pb-cdunn** [comment]
I'm curious: Why does htslib need to be in the PREFIX/src directory?

**Comment by mjhsieh** [comment]
because we have two htslib installed....

File: ports/pacbio/blasr/Makefile (hunk 1 above)
Comment (mjhsieh) [inline ports/pacbio/blasr/Makefile]: OSX does allow static linking some libs. The reason I used libhts.a is probably that I don't want to mess with (official) htslibs that comes with samtools and vice versa. But I guess samtools is more likely to be in danger in this mess.

**Comment by pb-cdunn** [comment]
Really? Within pitchfork we need 2 versions of htslib?

**Comment by mjhsieh** [comment]
basically pbbam uses pacbio branched htslib, while installing that version of htslib into prefix will screwup samtools building, since samtools bundled its own htslib. Oh crap, we have three htslibs in the system. Pysam also brings its own htslib...

**Comment by pb-cdunn** [comment]
Uh, oh. You're not gonna like to hear this, but bundled libraries do not belong in the PREFIX directory.

You can have a build-system-within-build-system. We're already doing that with **cmake**. Just build samtools independently, or distribute binaries. pitchfork should concentrate on flexibility only for packages we develop. (And I really think **cmake** is overkill for those.)

**Comment by mhsieh** [comment]
Previously only one version of the htslib (pacbio) was installed in the PREFIX, it's possible it got installed before samtools and pysam build and interfering with them. For example, the order of include argument in `$CFLAGS` would probably screwup stuff. I was puzzled last night for 3 hours trying to figure out some unrelated change could possible screw up my samtools build while I was using `make -j ` to use many threads.

**Comment by mhsieh** [comment]
That would be the reason why it was installed to the src/ at the first place. There is a second reason for that, I was not sure about some of the header files that came with the installation.

**Comment by mhsieh** [comment]
So although we have mentioned 3 copies of htslib, I think we only install one version (pacbio branch) and there won't be any mess-up in a sense of runtime.

**Pull request merged**